src/d3a/d3a_core/user_profile_handler.py

```diff
@@ -161,9 +162,16 @@ def _buffer_all_profiles(self, current_timestamp: DateTime):
                                                    self.convert_pendulum_to_datetime(end_time))
 
         for profile_uuid in self._profile_uuids:
-            self._user_profiles[profile_uuid] = \
-                {instance(data_point.time, TIME_ZONE): data_point.value
-                 for data_point in query_ret_val if data_point.profile_uuid == profile_uuid}
+            self._user_profiles[profile_uuid] = {
+                self.strip_timezone_and_create_pendulum_instance_from_datetime(
+                    data_point.time): data_point.value
+                for data_point in query_ret_val if data_point.profile_uuid == profile_uuid
+            }
+
+        for profile_uuid in self._user_profiles:
+            if not self._user_profiles[profile_uuid]:
+                self._user_profiles[profile_uuid] = self.get_first_week_from_profile(
+                    profile_uuid, current_timestamp)
```

Comment:
So I get that this is for the case when the time range (start_date, end_date) is not included in the DB, right? Shall we save the effort performing this step in the `_read_new_datapoints_from_buffer_or_rotate_profile` or shall we keep it as a safety net (because it is not that expensive) . Just double checking your rationale. Thanks

Reply:
Exactly, and if you do not mind I would really like to leave this as an extra safety. At the current state of the code the second database call in read_new_datapoints_from_buffer_or_update_profile was not called during my tests, but it does not hurt to have and can save us from hours of debugging